### PR TITLE
Method for getting compact from uint256

### PR DESCRIPTION
**Pull request by popzxc**
`BlockHeader` currently contains method to convert target from compact form to `Uint256`, but doesn't have one to convert from `Uint256` to the compact.
This PR fixes it. Used algorithm is the same one that is used in [`bictoin`](https://github.com/bitcoin/bitcoin) repository.

This method is needed for calculating required work for block.

I think in the future it will be better to create a separate `Compact` type which implements `From` for conversion, but for now I added it like this to not break compatibility.

### CODE REVIEW

File: src/blockdata/block.rs [hunk 1]
```diff
@@ -94,8 +94,28 @@ impl BlockHeader {
...
+            let bn = *value >> (8 * (size - 3));
+            bn.low_u32()
+        };
+
+        if (compact & 0x00800000) != 0 {
+            compact >>= 8;
+            size += 1;
+        }
+
+        assert!((compact & !0x007fffff) == 0);
```
Comment (tamasblummer) [inline src/blockdata/block.rs]: no panics in library code please. Avoid or propagate errors via Result<>

**Comment by popzxc** [comment]
Fixed. Those asserts were derived from [bitcoin/bitcoin](https://github.com/bitcoin/bitcoin) and are not really necessary.

**Comment by apoelstra** [comment]
Please squash.

**Comment by popzxc** [comment]
@apoelstra Github allows to squash&merge in one click, doesn't it? http://i.imgur.com/ee2CP1V.png

**Comment by tamasblummer** [comment]
@popzxc it is not about master history. Please squash as @apoelstra prefers squashed PRs for review.

**Comment by popzxc** [comment]
Ah, ok then.

**Comment by popzxc** [comment]
Done.

**Comment by apoelstra** [comment]
@popzxc it is not that I prefer "squashed history", it's that I prefer histories that make sense. That is, commits should be self-contained and not undo things from previous commits. (The exception is when adding extra commits to fix reviewer comments, which are useful during review, though I still prefer they be squashed into the original history once the review is done).

**Comment by popzxc** [comment]
@apoelstra so, can you please tell me what exactly should I do? I don't really understand it.

**Comment by apoelstra** [comment]
@popzxc In this case I'd just like you to squash both commits into one. But in general, before submitting PRs, it's good to use `rebase -i` to go over the list of commits and rearrange/squash them into a coherent story.

**Comment by tamasblummer** [comment]
@apoelstra PRs will need update as soon as an other PR pending was merged to master. If several PRs are outstanding we enter a tedious loop here. Update branch before review and squash merge should instead solve this in seconds.

**Comment by tamasblummer** [comment]
The same applies at this moment to https://github.com/rust-bitcoin/rust-bitcoin/pull/89: The merge master commit introduces zero changes. We could iterate a rebase or be more pragmatic?

**Comment by tamasblummer** [comment]
Thinking further I think @apoelstra is right.  Reviewer should not have to invest more than absolutely necessary in the process: If a 'merge master' commit is already there then PR submitter should rebase. I will do that on my PRs.

**Comment by popzxc** [comment]
Ok, if you say so.

**Comment by apoelstra** [comment]
@tamasblummer looks like merging is blocked regardless on some github thing related to your review. Can you clear that?

**Comment by apoelstra** [comment]
I think we should disable whatever option is forcing is to have everything built on the latest `master` when we merge PRs. That's dumb and redundant, you can already access the merged commits from upstream-pull/##/merge.

**Comment by tamasblummer** [comment]
I could merge after approving changes.

**Pull request merged**